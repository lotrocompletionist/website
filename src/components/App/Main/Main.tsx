import * as React from "react";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";
import * as Loadable from "react-loadable";
import { Header } from "@app/components/App/Header/Header";
import { Loading } from "@app/components/App/Loading/Loading";

const Instances = Loadable({
  loader: () => import("@app/components/WorldInstances/WorldInstances"),
  loading: Loading
});

const Raids = Loadable({
  loader: () => import("@app/components/Raids/Raids"),
  loading: Loading
});

const Skirmishes = Loadable({
  loader: () => import("@app/components/Skirmishes/Skirmishes"),
  loading: Loading
});

const VirtueDeeds = Loadable({
  loader: () => import("@app/components/VirtueDeeds/VirtueDeeds"),
  loading: Loading
});

const NotFound = Loadable({
  loader: () => import("@app/components/App/NotFound/NotFound"),
  loading: Loading
});

export const Main = () => (
  <Router>
    <>
      <Header />
      <Switch>
        <Redirect exact from="/" to="/instances" />
        <Route exact path="/instances" component={Instances} />
        <Route exact path="/raids" component={Raids} />
        <Route exact path="/skirmishes" component={Skirmishes} />
        <Route exact path="/virtue-deeds" component={VirtueDeeds} />
        <Route component={NotFound} />
      </Switch>
    </>
  </Router>
);
