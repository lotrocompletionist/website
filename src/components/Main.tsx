import * as React from "react";
import { Hello } from './Hello';

export const Main = () => <Hello title="Hello Schier!" />
