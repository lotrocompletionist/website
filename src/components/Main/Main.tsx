import * as React from "react";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";
import * as Loadable from "react-loadable";
import { Header } from "../Header/Header";
import { Loading } from "../Loading/Loading";

const Instances = Loadable({
  loader: () => import("../WorldInstances/WorldInstances"),
  loading: Loading
});

const Raids = Loadable({
  loader: () => import("../Raids/Raids"),
  loading: Loading
});

const Skirmishes = Loadable({
  loader: () => import("../Skirmishes/Skirmishes"),
  loading: Loading
});

const VirtueDeeds = Loadable({
  loader: () => import("../VirtueDeeds/VirtueDeeds"),
  loading: Loading
});

const NotFound = Loadable({
  loader: () => import("../NotFound/NotFound"),
  loading: Loading
});

export const Main = () => (
  <Router>
    <>
      <Header />
      <Switch>
        <Redirect exact from="/" to="/instances" />
        <Route exact path="/instances" component={Instances} />
        <Route exact path="/raids" component={Raids} />
        <Route exact path="/skirmishes" component={Skirmishes} />
        <Route exact path="/virtue-deeds" component={VirtueDeeds} />
        <Route component={NotFound} />
      </Switch>
    </>
  </Router>
);
