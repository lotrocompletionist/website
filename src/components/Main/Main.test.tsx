import * as React from "react";
import { create } from "react-test-renderer";
import { Main } from "./Main";

test("<Main> renders correctly", () => {
  const tree = create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
