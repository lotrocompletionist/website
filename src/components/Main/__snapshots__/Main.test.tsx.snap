// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main> renders correctly 1`] = `
<nav
  className="navbar Header is-link"
>
  <div
    className="navbar-brand"
  >
    <div
      className="navbar-item"
    >
      <img
        className="icon"
        src="icon.png"
      />
       LoTRO completionist
    </div>
    <div
      className="navbar-burger"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu"
    onClick={[Function]}
  >
    <div
      className="navbar-start"
    >
      <a
        className="navbar-item"
        href="/instances"
        onClick={[Function]}
      >
        Instances
      </a>
      <a
        className="navbar-item"
        href="/raids"
        onClick={[Function]}
      >
        Raids
      </a>
    </div>
  </div>
</nav>
`;
